### PR TITLE
POC V1: Home integration with none-ember apps

### PR DESCRIPTION
**Pull request by betaSocialCode**
@les2 @mvpc

This PR is just to have a discussion. Once an approach is agreed upon, new repo will be setup and the agreed upon solution will be implemented there. Will also need to setup deployment - jenkins etc.

The base of this PR has an empty/base ember app, so it's easier to review the real changes.
This POC was tested against: https://github.com/SocialCodeInc/advisor-frontend/tree/reporting-home-integration--POC-v1

### CODE REVIEW

**Comment by seanpmcb** [comment]
I dig it. But someone else should really weigh in for the review.

File: public/loader.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+(function() {
+
+  function createEl(sEl) {
+    return document.createElement(sEl);
+  }
+
+  function createScriptEl() {
+    var el = createEl('script');
+    el.async = false;
```
Comment (les2) [inline public/loader.js]: why do we make the scripts async false? Do we want to block the consuming app? maybe this is a good thing.

perhaps the loader should look for the settings object and use that to determine how aggressively to load the app shell.

File: public/loader.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  var fragment = document.createDocumentFragment();
+  var appJS = createScriptEl();
+  var vendorJS = createScriptEl();
+  var appCSS = createLinkEl();
+  var vendorCSS = createLinkEl();
+
+  appJS.src = '/home-shim/assets/home-shim.js';
+  vendorJS.src = '/home-shim/assets/vendor.js';
+  appCSS.href = '/home-shim/assets/home-shim.css';
+  vendorCSS.href = '/home-shim/assets/vendor.css';
```
Comment (les2) [inline public/loader.js]: how do we get the fingerprinted asset names in this file?

**Comment by les2** [comment]
can we get a little basic demo app within the repo that demos this?

btw, this is awesome! i think this has moved from POC to *this is what we're gonna do*

File: public/loader.js (hunk 2 above)
Comment (betaSocialCode) [inline public/loader.js]: The build takes care of it just like any other ref to assets

File: public/loader.js (hunk 1 above)
Comment (betaSocialCode) [inline public/loader.js]: The loader itself does not have async false. The scripts it loads have async false to ensure order of execution.

**Pull request merged**